packages/js/src/components/contentAnalysis/Results.js

```diff
@@ -236,6 +284,32 @@ class Results extends Component {
 		window.YoastSEO.analysis.applyMarks( new Paper( "", {} ), [] );
 	}
 
+	/**
+	 * Renders the modal for the highlighting upsell.
+	 *
+	 * @param {boolean} isOpen Whether the modal should be opened.
+	 * @param {function} closeModal A callback function invoked when the modal is closed.
+	 * @returns {wp.Element} The modal for the highlighting upsell element.
+	 */
+	renderHighlightingUpsell( isOpen, closeModal ) {
+		const upsellDescription = __(
+			"Highlight areas of improvement in your text, no more searching for a needle in a haystack, straight to optimizing! Now also in Elementor!",
+			"wordpress-seo" );
+
+		return ( isOpen && <Modal
+			title={ __( "Unlock Premium SEO analysis", "wordpress-seo" ) }
+			onRequestClose={ closeModal }
+			additionalClassName=""
+			className={ `${ defaultModalClassName } yoast-gutenberg-modal__box yoast-gutenberg-modal__no-padding` }
+			id="yoast-premium-seo-analysis-modal"
+			shouldCloseOnClickOutside={ true }
+		>
+			<ModalSmallContainer>
+				<PremiumSEOAnalysisUpsell buyLink={ this.props.highlightingUpsellLink } description={ upsellDescription } />
```

Comment:
Here we're reusing the component for the Premium SEO Analysis upsell since the only difference in the design is the description